### PR TITLE
Filter the SAM Runtimes list for Rider

### PR DESCRIPTION
**Pull request by hunterwerlla**
## Description

Issue: Rider supports Node, our Intellij plugin supports node, but our Rider plugin does not support Node. So, Node shows up in the list of runtimes but generates a C# project.

To fix this, add a filter to `SamInitSelectionPanel` which is where the runtime list is actually generated.

Other approaches considered: 
- Filter the list after the fact in `initSamPanel`. This seems less ideal because it fixes the problem after the fact.
- Implement Node support for Rider. I think this should be done in the future, but all of the existing code makes assumptions that it's C# in a csproj in a sln, so it would require major refactoring.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Screenshots (if appropriate)
Screen Shot 2019-10-31 at 8 21 04 AM

## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `gradlew check` succeeds
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/wizard/SamInitSelectionPanel.java [hunk 1]
```diff
@@ -42,7 +42,20 @@
...
     private final SamProjectGenerator generator;
 
+    /**
+     * Takes a runtime name and filters based on it.
+     */
+    private interface RuntimeFilter {
+        boolean filterExpression(String runtime);
+    }
+
+    private static final RuntimeFilter filterNoRuntimes = (s) -> true;
```
Comment (awschristou) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/ui/wizard/SamInitSelectionPanel.java]: Consider renaming to `includeAllRuntimes` to increase clarity by removing a negative from the name.

Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/wizard/SamInitSelectionPanel.java]: That's a better name, it used to be `filterNone` which is even less clear. Also added a changelog entry.

File: jetbrains-rider/src/software/aws/toolkits/jetbrains/ui/wizard/DotNetSamProjectGenerator.kt [hunk 2]
```diff
@@ -56,7 +57,7 @@ class DotNetSamProjectGenerator(
     )
 
     private val generator = SamProjectGenerator()
-    private val samPanel = SamInitSelectionPanel(generator)
+    private val samPanel = SamInitSelectionPanel(generator) { it.toLowerCase().startsWith("dotnetcore") }
```
Comment (abrooksv) [inline jetbrains-rider/src/software/aws/toolkits/jetbrains/ui/wizard/DotNetSamProjectGenerator.kt]: Query the `RuntmieGroup.DOTNET` so we dont have this logic that is different than elsewhere

Comment (hunterwerlla) [reply jetbrains-rider/src/software/aws/toolkits/jetbrains/ui/wizard/DotNetSamProjectGenerator.kt]: Good idea, fixed

**Pull request merged**